python/pyspark/pandas/groupby.py

```diff
@@ -2673,7 +2682,7 @@ def get_group(self, name: Union[Name, List[Name]]) -> FrameLike:
 
         return self._cleanup_and_return(DataFrame(internal))
 
-    def median(self, numeric_only: bool = True, accuracy: int = 10000) -> FrameLike:
+    def median(self, numeric_only: Optional[bool] = True, accuracy: int = 10000) -> FrameLike:
```

Comment:
Optional means that it can be `None` IIRC. Is that correct?

Reply:
Looks like it is right.

But the different is Pandas are using [`None` as default value](https://github.com/pandas-dev/pandas/blob/4bfe3d07b4858144c219b9346329027024102ab6/pandas/core/groupby/groupby.py#L1966), and use [_resolve_numeric_only](https://github.com/pandas-dev/pandas/blob/4bfe3d07b4858144c219b9346329027024102ab6/pandas/core/groupby/groupby.py#L1207) as a resolve to set default value (`true` or `false`).

@xinrong-databricks kept behaviors according pandas doc, could you also take a look, should we also add some logic like this?

Reply:
- DataFrameGroupBy
numeric_only=True/False/None -converts to-> numeric_only=True

In pandas, when numeric_only is False/None, a FutureWarning will be raised, and then numeric_only is converted to True.

**So I think setting True as the default value in pandas-on-Spark for numeric_only makes sense in this case**.

- SeriesGroupBy
numeric_only=None -converts to-> numeric_only=False

**So setting False as the default value in pandas-on-Spark for numeric_only seems fine**.

How do you think about that? @HyukjinKwon @Yikun

Reply:
I think it's ok for me, and you have added the regression test for `SeriesGroupBy` we can easy to catch this change if pandas changes these logic.

Reply:
I think it's ok for me, and we have already added regresion for `SeriesGroupBy`, we can easy to catch these logic changes in future. Thanks for explanation!

Reply:
Sorry I edited the explanation for SeriesGroupBy. Let me know if you have other concerns :).